plugins/modules/pnp_intent.py

```diff
@@ -945,6 +963,118 @@ def get_diff_deleted(self):
+            retrieving the current state (have) and desired state (want) of the configuration,
+            logs the states, and validates whether the specified device(s) exists in the DNA
+            Center configuration's PnP Database.
+        """
+
+        self.log(str(self.have))
+        self.log(str(self.want))
+        # Code to validate dnac config for merged state
+        device_response = self.dnac_apply['exec'](
+            family="device_onboarding_pnp",
+            function='get_device_list',
+            params={"serial_number": self.want.get("serial_number")}
+        )
+
+        if len(self.want.get("pnp_params")) == 1:
+            if not self.want["site_name"]:
+                if len(device_response) == 1:
+                    self.status = "success"
+                    msg = (
+                        "Requested Device with Serial No. {0} is "
+                        "present in Cisco DNA Center and "
+                        "addition verified.".format(self.want.get("serial_number")))
+                    self.log(msg)
+
+        elif len(self.want.get("pnp_params")) > 1:
+            devices_added = []
+            for device in self.want.get("pnp_params"):
+                multi_device_response = self.dnac_apply['exec'](
+                    family="device_onboarding_pnp",
+                    function='get_device_list',
+                    params={"serial_number": device["deviceInfo"]["serialNumber"]}
+                )
+                if (multi_device_response and (len(multi_device_response) == 1)):
+                    msg = (
+                        "Requested Device with Serial No. {0} is "
+                        "present in Cisco DNA Center and"
+                        " addition verified.".format(device["deviceInfo"]["serialNumber"]))
+                    self.log(msg)
+                else:
+                    msg = (
+                        "Requested Device with Serial No. {0} is "
+                        "not present in Cisco DNA "
+                        "Center".format(device["deviceInfo"]["serialNumber"]))
+                self.status = "success"
+
+        return self
+
+    def verify_diff_deleted(self, config):
+        """
+        Verify the deletion status of PnP configuration in Cisco DNA Center.
+        Args:
+            - self (object): An instance of a class used for interacting with Cisco DNA Center.
+            - config (dict): The configuration details to be verified.
+        Return:
+            - self (object): An instance of a class used for interacting with Cisco DNA Center.
+        Description:
+            This method checks the deletion status of a configuration in Cisco DNA Center.
+            It validates whether the specified device(s) exists in the DNA Center configuration's
+            PnP Database.
+        """
+
+        self.log(str(self.have))
+        self.log(str(self.want))
+        # Code to validate dnac config for merged state
```

Comment:
Same as like for merged.. Can we optimize/rewrite the below code?

```
    # Code to validate Catalyst Center config for deletion state
    for device_info in self.want.get("pnp_params", []):
        serial_number = device_info["deviceInfo"]["serialNumber"]
        
        device_response = self.dnac_apply['exec'](
            family="device_onboarding_pnp",
            function='get_device_list',
            params={"serial_number": serial_number}
        )

        if len(self.want.get("pnp_params")) == 1:
            if not self.want["site_name"] and not len(device_response) == 1:
                # Device is not present in Cisco Catalyst Center, deletion verified
                self.status = "success"
                msg = f"Requested Device with Serial No. {serial_number} is not presented in Cisco Catalyst Center. Deletion verified."
                self.log(msg)
        elif len(self.want.get("pnp_params")) > 1:
            multi_device_response = self.dnac_apply['exec'](
                family="device_onboarding_pnp",
                function='get_device_list',
                params={"serial_number": serial_number}
            )
            if not (multi_device_response and len(multi_device_response) == 1):
                # Device is not presented in Cisco Catalyst Center
                msg = f"Requested Device with Serial No. {serial_number} is not present in Cisco Catalyst Center. Deletion verified."
                self.log(msg)
            else:
                # Device is present in Cisco DNA Center
                msg = f"Requested Device with Serial No. {serial_number} is presented in Cisco Catalyst Center."
                self.status = "success"
                self.log(msg)

    return self
```

Reply:
Addressed